### PR TITLE
UI/Truncate long secret names

### PR DESCRIPTION
**Pull request by hellobontempo**
Fixes overflow problem of long secret key values mentioned in #12962

**Previous behavior:**
image

**Current behavior:**

truncate-long-labels

### CODE REVIEW

**Comment by ivana-mcconnell** [comment]
👍 The tooltip behaves as expected, but is it possible to have the pointer centered on the name of the key, rather than at the end of the name?

File: ui/lib/core/stories/info-table-row.stories.js [hunk 1]
```diff
@@ -16,14 +16,18 @@ storiesOf('InfoTable/InfoTableRow', module)
...
+        @isTooltipCopyable={{isTooltipCopyable}} />
     `,
       context: {
         label: text('Label', 'TTL'),
         value: text('Value', '30m (hover to see the tooltip!)'),
         helperText: text('helperText', 'This is helperText - for a short description'),
         alwaysRender: boolean('Always render?', false),
+        defaultShown: text('Default value', 'Some default value'),
         tooltipText: text('tooltipText', 'This is tooltipText'),
+        isTooltipCopyable: boolean('Allow tooltip to be copied', true),
```
Comment (zofskeez) [inline ui/lib/core/stories/info-table-row.stories.js:30]: Thanks for adding this for me!

File: ui/lib/core/addon/components/info-table-row.js [hunk 2]
```diff
@@ -63,4 +64,16 @@ export default Component.extend({
...
+
+  didInsertElement() {
+    this._super(...arguments);
+    const labelDiv = this.element.querySelector('div');
+    const labelText = this.element.querySelector('.is-label');
+    if (labelDiv && labelText) {
+      if (labelText.offsetWidth > labelDiv.offsetWidth) {
+        labelDiv.classList.add('label-overflow');
+        this.set('hasLabelOverflow', true);
+      }
```
Comment (zofskeez) [inline ui/lib/core/addon/components/info-table-row.js:76]: Nice work! `didInsertElement` is a good place for this. I'm wondering if it is worthwhile to add a test to check if the `label-overflow` class is applied to the element and that the tooltip appears for long labels?

File: ui/lib/core/addon/components/bar-chart.js [hunk 3]
```diff
@@ -19,7 +19,7 @@
  * @param {string} title - title of the chart
  * @param {array} mapLegend - array of objects with key names 'key' and 'label' for the map legend
  * @param {object} dataset - dataset for the chart
- * @param {array} tooltipData - misc. information needed to display tooltip (i.e. total clients from query params)
+ * @param {any} tooltipData - misc. information needed to display tooltip (i.e. total clients from query params)
```
Comment (Monkeychip) [inline ui/lib/core/addon/components/bar-chart.js:22]: I'm not sure I've seen every before. Does it come in as an object, or varies? Maybe describe / give example of types in the description?

**Review by Monkeychip — APPROVED** [review]
Nice. At first I was wondering about a test, but then I'm not sure if it's super relevant. I'll leave it up to you.

**Review by zofskeez — COMMENTED** [review]
Nice work and thanks again for adding the _isTooltipCopyable_ property to the InfoTableRow story that I forgot!

File: ui/lib/core/addon/components/bar-chart.js (hunk 3 above)
Comment (hellobontempo) [reply ui/lib/core/addon/components/bar-chart.js:22]: @Monkeychip I added it after seeing it somewhere else...but yeah. Going to open a separate PR that will address some tweaks this file needs.

File: ui/lib/core/addon/components/info-table-row.js (hunk 2 above)
Comment (hellobontempo) [reply ui/lib/core/addon/components/info-table-row.js:76]: Thanks @zofskeez! I'd gone back and forth on a test..but I think that you're right to suggest it- this component shows up so many places it's probably worthwhile to include. I'll add!

**Pull request merged**